rotkehlchen/chain/manager.py

```diff
@@ -824,384 +829,49 @@ def sync_bitcoin_accounts_with_db(
             if account not in db_btc_accounts:
                 accounts_to_remove.append(account)
 
-        if blockchain == SupportedBlockchain.BITCOIN:
-            balances_mapping = get_bitcoin_addresses_balances(accounts_to_remove)
-        else:
-            balances_mapping = get_bitcoin_cash_addresses_balances(accounts_to_remove)
-        balances = [balances_mapping.get(x, ZERO) for x in accounts_to_remove]
         self.modify_blockchain_accounts(
             blockchain=blockchain,
             accounts=accounts_to_remove,
             append_or_remove='remove',
-            add_or_sub=operator.sub,
-            already_queried_balances=balances,
-        )
-
-    def modify_btc_account(
-            self,
-            account: BTCAddress,
-            append_or_remove: str,
-            add_or_sub: AddOrSub,
-            already_queried_balance: Optional[FVal] = None,
-    ) -> None:
-        """Either appends or removes a BTC acccount.
-
-        If already_queried_balance is not None then instead of querying the balance
-        of the account we can use the already queried one.
-
-        Call with 'append', operator.add to add the account
-        Call with 'remove', operator.sub to remove the account
-
-        May raise:
-        - RemotError if there is a problem querying blockchain.info or cryptocompare
-        """
-        btc_usd_price = Inquirer().find_usd_price(A_BTC)
-        remove_with_populated_balance = (
-            append_or_remove == 'remove' and len(self.balances.btc) != 0
-        )
-        # Query the balance of the account except for the case when it's removed
-        # and there is no other account in the balances
-        if append_or_remove == 'append' or remove_with_populated_balance:
-            if already_queried_balance is None:
-                balances = get_bitcoin_addresses_balances([account])
-                balance = balances[account]
-            else:
-                balance = already_queried_balance
-            usd_balance = balance * btc_usd_price
-
-        if append_or_remove == 'append':
-            self.balances.btc[account] = Balance(amount=balance, usd_value=usd_balance)
-        elif append_or_remove == 'remove':
```

Comment:
In discord you mentioned 

> The bug was actually that balances and totals were not reset when a query was called and thus data from the blockchain accounts that were deleted was still returned


And as I said that's not the bug. The bug is you went and removed a lot of code without understanding what you are removing. 

In all the modify functions, if the modification is about a removal the balances mapping is cleaned for the address. And you removed this cleaning. This is the bug.

Fix it by keeping that functionality for all of them.  All the removal functionality that is. For all the chains.

Reply:
Kept the removal functionality